docs/reference/plugins/kernel_plugin.rst

```diff
@@ -0,0 +1,198 @@
+.. _reference-kernel-plugin:
+
+Kernel plugin
+==============
+
+The Kernel plugin simplifies the building of kernel snaps and kernel modules.
+It helps with both Ubuntu-specific kernels as well as kernels from some
+arbitrary upstream source, such as the mainline or linux-next trees.
+
+The kernel build process tends to be highly repetitive and deterministic, where
+most desired modifications include applying patches and modifying Kconfig
+values. This plugin allows for the easy toggling of Kconfig values and entire
+defconfigs used during builds to improve kernel snap iterability and
+maintainability.
+
+
+Keys
+----
+
+This plugin provides the following unique keys.
+
+
+kernel-kdefconfig
+~~~~~~~~~~~~~~~~~
+
+**Type**: list of strings
+
+**Default**: ``["defconfig"]``
+
+The kernel configuration(s) to use when generating a ``.config``, such as those
+found in ``arch/${arch}/configs`` or ``kernel/configs``.
+
+
+kernel-kconfigflavour
+~~~~~~~~~~~~~~~~~~~~~
+
+**Type**: list of strings
+
+**Default**: ``["generic"]``
+
+The Ubuntu kernel flavor to use when generating an Ubuntu kernel config. Valid
+names are things like ``lowlatency`` and ``generic``. A list of some variants
+can be found `here <https://ubuntu.com/kernel/variants>`_, but valid choices
+will depend on the chosen kernel source. If something other than ``generic`` is
+provided, this option wins over ``kernel-kdefconfig``.
+
+
+kernel-kconfigs
+~~~~~~~~~~~~~~~
+
+**Type**: list of strings
+
+**Default**: ``[]``
+
+An explicit list of Kconfig values to force; this will override the configs
+that are set in the base configuration through ``kernel-kdefconfig`` and
+``kernel-kconfigflavour``. Dependent Kconfigs will be fixed using the kbuild
+system to regenerate a valid configuration.
+
+
+kernel-enable-zfs-support
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**Type**: bool
+
+**Default**: ``false``
+
+Enabling this option will build the ZFS kernel modules alongside the kernel.
+
+
+kernel-enable-perf
+~~~~~~~~~~~~~~~~~~
+
+**Type**: bool
+
+**Default**: ``false``
+
+Enabling this option will build the ``perf`` binary.
+
+
+Environment variables
+---------------------
+
+This plugin responds to its own special variables.
```

Comment:
```suggestion
This plugin provides the following environment variables.
```

Reply:
Damn you, [unreliable prior art](https://documentation.ubuntu.com/snapcraft/latest/reference/plugins/uv_plugin/#environment-variables)!